juntagrico/management/commands/generate_testdata_advanced.py

```diff
@@ -110,63 +114,72 @@ def get_random_props(self):
         props = self.get_random_props()
         mem1_fields = self.generate_member_dict(props)
         main_member = Member.objects.create(**mem1_fields)
         self.generate_shares(main_member, sub_shares)
         mem2_fields = self.generate_member_dict(props)
         co_member = Member.objects.create(**mem2_fields)
         self.generate_shares(main_member, sub_shares)
-        self.generate_subscription(main_member, co_member, depot, type)
+        self.generate_subscription(main_member, co_member, depot, sub_types)
 
     def add_arguments(self, parser):
-        parser.add_argument('--sub-size', type=int, help='Size of subscription', default=2)
+        parser.add_argument('--products', type=str, help='Products to use', default=["Gemüse"], nargs="+")
+        parser.add_argument('--sub-size', type=int, help='Size of subscription', default=[1], nargs="+")
         parser.add_argument('--sub-shares', type=int, help='Required shares per subscription', default=2)
         parser.add_argument('--sub-assignments', type=int, help='Required assignment per subscription', default=6)
-        parser.add_argument('--sub-prize', type=int, help='Proce of suscription', default=250)
+        parser.add_argument('--sub-price', type=int, help='Price of suscription', default=250)
         parser.add_argument('--depots', type=int, help='Number of depots to generate', default=4)
+        parser.add_argument('--days', type=int, help='Days for depots', default=list(range(7)), nargs="+")
         parser.add_argument('--subs-per-depot', type=int, help='Subscriptions per depot', default=10)
         parser.add_argument('--job-amount', type=int, help='Jobs per area', default=10)
 
     # entry point used by manage.py
     def handle(self, *args, **options):
-        subprod_field = {'name': 'Gemüse'}
-        sub_product, created = SubscriptionProduct.objects.get_or_create(**subprod_field)
-
-        subsize_fields = {'name': 'Normales Abo', 'long_name': 'Ganz Normales Abo', 'units': options['sub_size'],
-                          'depot_list': True,
-                          'description': 'Das einzige abo welches wir haben, bietet genug Gemüse für einen Zwei personen Haushalt für eine Woche.',
-                          'product': sub_product}
-        size, _ = SubscriptionSize.objects.get_or_create(**subsize_fields)
-
-        subtype_fields = {
-            'name': 'Normales Abo typ',
-            'long_name': '',
-            'shares': options['sub_shares'],
-            'required_assignments': options['sub_assignments'],
-            'price': options['sub_prize'],
-            'description': '',
-            'size': size
-        }
-        type, _ = SubscriptionType.objects.get_or_create(
-            name=subtype_fields['name'],
-            defaults=subtype_fields
-        )
+        sub_types = []
+        for product, size in itertools.zip_longest(options['products'], options['sub_size']):
```

Comment:
This fails, if there are more sub_size than products, because products is then `None` for the last iterations.

Reply:
Redone it differently in a different pull-request, based all changes on the release_1.6 branch this time. check #659